### PR TITLE
Avoid crash if imported in a web worker

### PR DESCRIPTION
**Pull request by Pessimistress**

### CODE REVIEW

File: src/utils/hammer.js [hunk 1]
```diff
@@ -18,19 +18,4 @@
...
-  instance.on = chainedNoop;
-  instance.off = chainedNoop;
-  instance.destroy = chainedNoop;
-  instance.emit = chainedNoop;
-  return instance;
-}
-
-export const Manager = HammerManagerMock;
-
 export default null;
```
Comment (ericsoco) [inline src/utils/hammer.js]: I guess unrelated to this diff, but why explicitly export default null?

File: src/event-manager.js [hunk 2]
```diff
@@ -42,7 +43,7 @@ const DEFAULT_OPTIONS = {
   // custom recognizers
   recognizers: null,
   // Manager class
-  Manager,
+  Manager: Hammer ? Hammer.Manager : HammerManagerMock,
```
Comment (ericsoco) [inline src/event-manager.js]: Looking at the code in this PR, seems like `Hammer` is always `null`??
Are you somehow swapping `hammer.browser.js` in for `utils/hammer.js` via Webpack or something? You should make this strategy clear in a comment along with the logic that gets one or the other module at runtime (i.e. here).

File: src/utils/hammer.browser.js [hunk 3]
```diff
@@ -1,9 +1,32 @@
...
+// THE SOFTWARE.
+
 import {enhancePointerEventInput, enhanceMouseInput} from './hammer-overrides';
 
-enhancePointerEventInput(hammerjs.PointerEventInput);
-enhanceMouseInput(hammerjs.MouseInput);
+let hammerjs = null;
 
-export const Manager = hammerjs.Manager;
+if (typeof document !== 'undefined') {
```
Comment (ericsoco) [inline src/utils/hammer.browser.js]: Nit: Any reason not to just `if (!document)`?

File: src/event-manager.js (hunk 2 above)
Comment (Pessimistress) [reply src/event-manager.js]: Yes, using the `browser` field in package.json. I will add a comment.

File: src/utils/hammer.browser.js (hunk 3 above)
Comment (Pessimistress) [reply src/utils/hammer.browser.js]: Will throw `ReferenceError: document is not defined`

File: src/utils/hammer.js (hunk 1 above)
Comment (Pessimistress) [reply src/utils/hammer.js]: This was originally an attempt to separate Node-only code and browser-only code. Now that the browser part does not work in all cases it does seem redundant. :(

File: src/event-manager.js (hunk 2 above)
Comment (danmarshall) [reply src/event-manager.js]: Does this strictly limit usage to only Webpack? I am using the Parcel bundler.

File: src/utils/hammer.browser.js [hunk 4]
```diff
@@ -1,9 +1,32 @@
...
+  // Avoid crash if imported in a web worker
+  hammerjs = require('hammerjs');
```
Comment (TrySound) [inline src/utils/hammer.browser.js]: Hey @Pessimistress. I just upgraded and it broke my build again. Please NEVER mix commonjs and esm. It's webpackism which is not handled everywhere else well.
My change was intended to fix this. And you reverted it back.
https://github.com/uber-web/mjolnir.js/pull/22

Comment (TrySound) [reply src/utils/hammer.browser.js]: Why this module is run inside of worker?

Comment (Pessimistress) [reply src/utils/hammer.browser.js]: 🤦🏻‍♀️sorry about that. This was thrown in a while ago when we experimented with running deck.gl on a webworker.

I will introduce eslint-plugin-import to block dynamic require.

Comment (TrySound) [reply src/utils/hammer.browser.js]: So you can just revert to my previous solution, right?

**Pull request merged**